planning/behavior_path_sampling_planner_module/src/sampling_planner_module.cpp

```diff
@@ -0,0 +1,981 @@
+// Copyright 2023 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "behavior_path_sampling_planner_module/sampling_planner_module.hpp"
+
+namespace behavior_path_planner
+{
+using geometry_msgs::msg::Point;
+using motion_utils::calcSignedArcLength;
+using motion_utils::findNearestIndex;
+using motion_utils::findNearestSegmentIndex;
+using tier4_autoware_utils::calcDistance2d;
+using tier4_autoware_utils::calcOffsetPose;
+using tier4_autoware_utils::getPoint;
+using tier4_autoware_utils::Point2d;
+
+namespace bg = boost::geometry;
+namespace bgi = boost::geometry::index;
+
+SamplingPlannerModule::SamplingPlannerModule(
+  const std::string & name, rclcpp::Node & node,
+  const std::shared_ptr<SamplingPlannerParameters> & parameters,
+  const std::unordered_map<std::string, std::shared_ptr<RTCInterface>> & rtc_interface_ptr_map,
+  std::unordered_map<std::string, std::shared_ptr<ObjectsOfInterestMarkerInterface>> &
+    objects_of_interest_marker_interface_ptr_map)
+: SceneModuleInterface{name, node, rtc_interface_ptr_map, objects_of_interest_marker_interface_ptr_map},  // NOLINT
+  vehicle_info_{vehicle_info_util::VehicleInfoUtil(node).getVehicleInfo()}
+{
+  internal_params_ =
+    std::shared_ptr<SamplingPlannerInternalParameters>(new SamplingPlannerInternalParameters{});
+  updateModuleParams(parameters);
+
+  // check if the path is empty
+  hard_constraints_.emplace_back(
+    [](
+      sampler_common::Path & path, [[maybe_unused]] const sampler_common::Constraints & constraints,
+      [[maybe_unused]] const MultiPoint2d & footprint) -> bool {
+      return !path.points.empty() && !path.poses.empty();
+    });
+
+  hard_constraints_.emplace_back(
+    [](
+      sampler_common::Path & path, const sampler_common::Constraints & constraints,
+      const MultiPoint2d & footprint) -> bool {
+      if (!footprint.empty()) {
+        path.constraint_results.drivable_area =
+          bg::within(footprint, constraints.drivable_polygons);
+      }
+
+      for (const auto & f : footprint) {
+        const auto collision_index = constraints.rtree.qbegin(bgi::intersects(f));
```

Comment:
Please register to dictionary.
https://github.com/tier4/autoware-spell-check-dict/actions/workflows/add-word.yaml

Reply:
Done, thanks!